src/core/socket.c

```diff
@@ -1645,6 +1645,10 @@ static int socket_open_fds(Socket *orig_s) {
                         switch (p->address.type) {
 
                         case SOCK_STREAM:
+                                if (IN_SET(s->socket_protocol, IPPROTO_SCTP, IPPROTO_MPTCP))
+                                        p->address.protocol = s->socket_protocol;
+                                break;
+
                         case SOCK_SEQPACKET:
                                 if (s->socket_protocol == IPPROTO_SCTP)
                                         p->address.protocol = s->socket_protocol;
```

Comment:
You can fall through on SCTP here, i.e.:

```suggestion
                         case SOCK_STREAM:
                                if (s->socket_protocol == IPPROTO_MPTCP) {
                                        p->address.protocol = s->socket_protocol;
                                        break;
                                }
                                _fallthrough_;

                         case SOCK_SEQPACKET:
                                 if (s->socket_protocol == IPPROTO_SCTP)
                                         p->address.protocol = s->socket_protocol;
```

Reply:
(but doesn't matter really. If you think the current form is more readable feel free to keep it)

Reply:
@YHNdnzj thank you for this suggestion!

Personally, I find the current version more readable, especially if it is expected to accept more socket protocol later on. But I'm happy to change if this suggestion is preferred, the result is the same.

Reply:
I prefer @matttbe's code.